models/notification.go

```diff
@@ -311,3 +311,14 @@ func getNotificationByID(notificationID int64) (*Notification, error) {
 
 	return notification, nil
 }
+
+// SwapNotificationStatuses swaps the statuses of all of a user's notifications that are of the currentStatus type
+func SwapNotificationStatuses(user *User, currentStatus NotificationStatus, desiredStatus NotificationStatus) error {
```

Comment:
Maybe `UpdateNotificationStatuses` is a better name? My first thought when I saw the name was that it replaced all occurrences of status A with status B, __and__ replaced all occurrences of status B with status A.

Reply:
Sure, will refactor accordingly